File: 070/main.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
  int climbStairs(int n)
  {
    int arr[n + 2];
    arr[1] = 1;
    arr[2] = 2;
    for (int i = 3; i <= n; i++)
    {
      arr[i] = arr[i - 2] + arr[i - 1];
    }
    return arr[n];
  }
};

int main()
{
  Solution sol;
  int res = sol.climbStairs(5);
  cout << res << endl;
  return 0;
}

File: 070/main.js
var climbStairs = function (n) {
  let res = Array.from({length: n}).map(() => 0)
  res[1] = 1
  res[2] = 2
  for (let i = 3; i <= n; i++) {
    res[i] = res[i - 2] + res[i - 1]
  }
  return res[n]
}

console.log(climbStairs(2))
console.log(climbStairs(3))
